src/applications/mhv/medical-records/tests/actions/allergies.unit.spec.js

```diff
@@ -32,20 +35,78 @@ describe('Get allergies action', () => {
 });
 
 describe('Get allergy action', () => {
-  it('should dispatch a get details action', () => {
+  it('should dispatch a get details action', async () => {
+    const allergyList = [];
     const mockData = allergy;
+    const dispatch = sinon.spy();
+
+    const getDetail = async () => mockData;
+
+    try {
+      await dispatchDetails(
+        '3106',
+        allergyList,
+        dispatch,
+        getDetail,
+        Actions.Allergies.GET_FROM_LIST,
+        Actions.Allergies.GET,
+      );
+      expect(dispatch.called).to.be.true;
+      expect(dispatch.firstCall.args[0]).to.exist;
+      expect(dispatch.firstCall.args[0].type).to.equal(
+        Actions.Allergies.GET_FROM_LIST,
+      );
+    } catch (error) {
+      dispatch(addAlert(Constants.ALERT_TYPE_ERROR));
+    }
+  });
```

Comment:
This test is just calling the lower-level `dispatchDetails` function without ever calling the action function. You should have two detail-testing functions for each domain. You can put them in the same describe block since they both test the same function. They should look as follows (using conditions as a model):

```
describe('Get condition action', () => {
  it('should dispatch a get details action', () => {
    const mockData = condition;
    mockApiRequest(mockData);
    const dispatch = sinon.spy();
    return getConditionDetails('1', undefined)(dispatch).then(() => {
      expect(dispatch.firstCall.args[0].type).to.equal(Actions.Conditions.GET);
    });
  });

  it('should dispatch a get details action using the existing list', () => {
    const dispatch = sinon.spy();
    return getConditionDetails('1', [{ id: '1' }])(dispatch).then(() => {
      expect(dispatch.firstCall.args[0].type).to.equal(
        Actions.Conditions.GET_FROM_LIST,
      );
    });
  });
});
```